include/utils.h

```diff
@@ -106,19 +98,20 @@ MinimizerHitClusters get_minimizer_hit_clusters(
     const std::vector<std::string> &prg_names,
     std::shared_ptr<MinimizerHits> minimizer_hits,
     std::shared_ptr<pangenome::Graph> pangraph, const int max_diff,
-    const uint32_t& genome_size, const float& fraction_kmers_required_for_cluster,
+    const float& fraction_kmers_required_for_cluster,
     ClusterDefFile &cluster_def_file,
     ClusterFilterFile &cluster_filter_file,
     const uint32_t min_cluster_size,
-    const uint32_t expected_number_kmers_in_read_sketch = std::numeric_limits<uint32_t>::max());
+    const uint32_t expected_number_kmers_in_read_sketch,
+    const uint32_t rng_seed);
 
 uint32_t pangraph_from_read_file(const SampleData& sample,
     std::shared_ptr<pangenome::Graph> pangraph, Index &index,
     const int max_diff, const float& e_rate,
     const fs::path& sample_outdir, const uint32_t min_cluster_size = 10,
-    const uint32_t genome_size = 5000000, const bool illumina = false, const bool clean = false,
-    const uint32_t max_covg = 300, uint32_t threads = 1,
-    const bool keep_extra_debugging_files = false);
+    const uint32_t genome_size = 5000000, const uint32_t max_covg = 300,
```

Comment:
Is the default value if you call this function without specifying the `max_covg` argument, which we never do. We actually use the value passed by the `--max-covg` argument, which is also 300 if not given. If want to change this value, needs to call pandora with `--max-covg <value>`

Reply:
ok, well i have no problem with this PR, but we need to have a think about how default covgs are used

Reply:
In this function, it is used just to know when to stop processing reads: https://github.com/leoisl/pandora/blob/e0881e8d1e166303e46d41481f5b2f979f0dc63f/src/utils.cpp#L605-L612

e.g. if the user specified `--max-covg 100`, and the reads has 1000x depth, after we mapped 100x we stop

Reply:
this is fine. I was worried about plasmids. if you ask the user whats the max coverage, they think in terms of the chromosome and say 100x. But a 100 copy plasmid will have 10000x just over it. But you are using this as a global measure of all the data, and dividing by genome-size so it;s fine

Reply:
> But a 100 copy plasmid will have 10000x just over it.

I am confused if is an issue or not. This works in conjunction with the `--genome-size` parameter. If you say the `--genome-size` is 5MB and the `--max-covg` is 100, `pandora` will map 500 MB of bases. Once it mapped this amount of data, it will stop. In practice, for RH, I just set `--max-covg` to 1000000, so it always map all reads, and never downsample

Reply:
ah that's an interesting question. let's try a small calculation. 
chrom = 5Mb
plasmid 1 = 7kb, copy number 100;  total length of input genome=700kb
plasmid 2 = 20kb, copy number = 10; total length of input genome = 200kb
plasmid 3 = 200kb, copy number = 1; total length of input genome = 200kb
Total plasmid length including copy number is just over 1Mb = 1.1Mb

If we get 500MB reads, this will basically be 400MB for the chromosome, so actually 80x depth not 100x
For the plasmids, we will get 100Mb split over those plasmids, which will be approximately 100x of plasmid 1, and 10x of plasmid 2 and 1x of plasmid 3 

So it sort of works, but all our calculations about expecting 100x on the chromosomal genes will be wrong, we'll only get 80x. 

Not very easy to fix this properly though.

Reply:
Plasmid coverage is a bit indeed complicated, however we can simplify things by setting `--max-covg 10000000`. This will just tell pandora to map all reads. Then pandora will map all reads, and will check how many bases mapped and divide over the genome size (see https://github.com/leoisl/pandora/blob/e0881e8d1e166303e46d41481f5b2f979f0dc63f/src/utils.cpp#L668-L669). That will be the coverage that pandora will actually use, it won't use what was given to `--max-covg` as coverage. Genome size is 5MB by default. In the plasmid use case, the coverage will thus always be super low, because all chromosomal reads won't map, we will just map plasmid reads, and then we will still divide the cumulative size of the mapped reads by 5MB. This coverage is used to update the error rate to be used, the kmer model, set the minimum kmer coverage for genotyping, and to define which genes are FP (e.g. too low or high coverage). All these stats don't make much sense in the plasmid use case and we got better results in RH just by ignoring these (e.g. don't update error rate and kmer model, don't remove genes based on coverage, etc). That is why in a few previous PRs we have parameters to skip doing these things. So, in summary, we simply don't apply any coverage-related thresholding in pandora when running it through RH

Reply:
Yeah this is behaviour I have always advocated for removing. To me, it's the user's responsibility to downsample reads, not pandora. For the exact reasons stated here. I use `--max-covg 10000000` (or something equally huge) in drprg to disable this functionality.

Reply:
Makes sense, I am using the same value for `--max-covg` as you do, I think we then should remove it: https://github.com/rmcolq/pandora/issues/340